doc/source/guide/in_depth/functional_guide.rst

```diff
@@ -0,0 +1,257 @@
+.. _functional_in_depth:
+
+#####################
+Functional
+#####################
+
+A *functional* is an operator ``f`` that maps from some vector space ``X`` to the field of scalars ``F`` associated with the vector space:
+
+.. math::
+   
+   f : X \to F.
+
+In the ODL solver package, functionals are implemented in the `Functional` class, a subclass to `Operator`.
+
+From a mathematical presepective, the above is a valide definition of a functional.
+However, since the purpose of these functionals are primarely to be used for solving optimization problems, the following assumptions are made:
+
+ * the vector space ``X`` is a Hilbert space.
+ * the field of scalars ``F`` are the real numbers.
```

Comment:
This is intended. Most of the results used in the implementation should probably hold also in this case, but I cannot guarantee that right away (for example a linear perturbation of this functional does not make sense from an optimization perspective, but `proximal_quadratic_perturbation` does not check that it is a real space...). Therefore we have only gone for real vector spaces.

However, as mentioned below, there is no checking for this. So if you as a user would like to try the functional
```
|| . || C^n   ->   R
```
nothing is going to stop you. But we do not guarantee that everything works as expected.